package/gluon-core/files/lib/netifd/proto/gluon_wired.sh

```diff
@@ -16,7 +16,7 @@ xor2() {
 
 interface_linklocal() {
         local macaddr="$(ubus call network.device status '{"name": "'"$1"'"}' | jsonfilter -e '@.macaddr')"
-        local oldIFS="$IFS"; IFS=':'; set -- $macaddr; IFS="$oldIFS"
+        local oldIFS="$IFS"; IFS=':'; set -- "$macaddr"; IFS="$oldIFS"
```

Comment:
This change is incorrect. `set` must be called without quotes, so the bytes of the MAC address are assigned to separate arguments.

Reply:
oh yes obviously ...